lib/streamlit/runtime/state/widgets.py

```diff
@@ -101,6 +103,7 @@
         "text_input": "string_value",
         "time_input": "string_value",
         "component_instance": "json_value",
+        "data_editor": "string_value",
```

Comment:
Curious, why is this not JSON value?

Reply:
I used JSON in the beginning. It only adds a `JSON.stringify` on frontend and `json.loads` on the backend, but - if I remember correctly - it required some workarounds and changes for some situations. So, the abstraction offered by the json type wasn't really adding any benefits.